wp-includes/rest-api/endpoints/class-wp-rest-customize-changesets-controller.php

```diff
@@ -560,6 +561,157 @@ public function create_item( $request ) {
 	}
 
 	/**
+	 * Deletes a changeset.
+	 *
+	 * @since ?.?.?
+	 * @access public
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function delete_item( $request ) {
+		global $wpdb;
+
+		$manager = $this->ensure_customize_manager( $request['uuid'] );
+
+		if ( ! $manager->changeset_post_id() ) {
+			return new WP_Error(
+				'rest_post_invalid_uuid',
+				__( 'Invalid changeset UUID.' ),
+				array(
+					'status' => 404,
+				)
+			);
+		}
+
+		$post = get_post( $manager->changeset_post_id() );
+
+		if ( $request['force'] ) {
+			$previous = $this->prepare_item_for_response( $post, $request );
+
+			// TODO: At this point $wp_customize will no longer have up-to-date post data.
```

Comment:
We may want to consider updating `\WP_Customize_Manager::changeset_data()` to eliminate the `if ( isset( $this->_changeset_data ) ) {` check entirely so that `\WP_Customize_Manager::get_changeset_post_data()` will always be called. Either that, or we should add `delete_post` and `save_post_customize_changeset` (or better yet, `clean_post_cache`) actions in the constructor to unset `\WP_Customize_Manager::$_changeset_data`.

That being said, all of the changeset data should actually be abstracted into a separate class to make it easier to work with changesets irrespective of a manager. See https://core.trac.wordpress.org/ticket/40527

I'd recommend work to be done as part of that ticket.